vhost_user_block/src/lib.rs

```diff
@@ -522,6 +517,16 @@ pub fn start_block_backend(backend_command: &str) {
         process::exit(1);
     }
 
+    loop {
+        if blk_backend.read().unwrap().mem.is_some() && blk_backend.read().unwrap().vrings.is_some()
+        {
+            if let Err(e) = blk_backend.write().unwrap().run() {
+                error!("Failed to run worker thread for vhost-user-blk: {:?}", e);
+            }
+            break;
+        }
+    }
+
```

Comment:
This is a terribly ugly hack. If worker threads can't be created in advance for some reason, they must created after receiving a VHOST_USER_SET_VRING_KICK.

Reply:
> So I guess this is about avoiding the backend RwLock write lock at VringEpollHandler. That's fine and it's something that we needed to tackle sooner than later.
> 
> I've added some suggestions to the code. There are also a number of commits with a summary but empty description. I think that should be fixed too.

Thanks for helping review this and providing good suggestions, I have addressed them (except one comment to be added). PTAL. But, I see the event_idx issue is still there, do you have any clue? Why this happens?